### DISCUSSION TITLE
ICM shows Offline while connected to Internet - how to resolve

### DISCUSSION THREAD

**Discussion by dmishh** [InternetConnectionMonitor/icm-public#113] [Q&A] [answered]
The reason ICM shows **Offline** when you're connected to Internet - **another extension or software blocks ICM connections to Google**. It could be an anti-virus, ad blocker, malware scanner, etc.

For example, **Malwarebytes Browser Guard** extension blocks the ICM monitoring.
Their recent update in early August 2024 breaks the ICM connection to Google.

**Comment by dmishh** [comment, marked as answer]
You can **turn it off** or continue to use it adding an exception.

To add the exception in Malwarebytes Browser Guard and restore ICM monitoring do this:

**(1) Open "Allow list" in the Malwarebytes Browser Guard**

<img src="https://github.com/user-attachments/assets/91500165-77c6-4510-b195-ab75a1a604a7" width="350" />


**(2) Add an exception rule for "https://google.com" for all 3 types of detection**

<img src="https://github.com/user-attachments/assets/2e50485a-2439-42b4-b7d3-43da07192e40" width="350" />


**(3) ICM should work fine now**

**Comment by glg0608** [comment]
is there a way to get the old app that I can start and have a constant log on the connection back?

**Comment by toshim45** [comment]
Hi @dmishh 
I can't figure out why ICM offline, but my traceroute to 8.8.8.8 is success, also I dont have any extension that block icm. Could u pls help ?
I attached my screenshot. 
<img width="1134" height="660" alt="icm-chrome-offline" src="https://github.com/user-attachments/assets/69bcb8c7-2f5f-4783-b6d8-f1b5d1492fa3" />

**Reply by dmishh** [reply]
@toshim45 

Hey! So the thing is - traceroute to 8.8.8.8 only proves that ICMP packets reach the destination. But ICM checks connectivity via HTTP protocol, and http://8.8.8.8 doesn't actually serve any web content - it just doesn't respond on port 80/443.

So your traceroute can be perfectly fine while ICM still shows offline, because they test different things.

To test what ICM actually tests, try running:

`wget -O /dev/null http://google.com`

or

`curl -I http://google.com`

If that fails - then it's likely something blocking HTTP traffic specifically (firewall, proxy, etc).

**Reply by toshim45** [reply]
Yesterday my issue was fixed, I didn't know what was my internet provider did.
Thank you for your reply, learned `curl -I` just now

<img width="884" height="445" alt="icm-fixed" src="https://github.com/user-attachments/assets/4aa6b870-9abe-4c76-b2c8-489c8095a4ed" />
